application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,213 @@
+
+
+        TextInput transferQuestionTitle =
+                TextInput.create(TRANSFER_QUESTION_TITLE_ID, "Title", TextInputStyle.SHORT)
+                    .setMaxLength(70)
+                    .setMinLength(8)
+                    .setValue(createTitle(originalMessage))
+                    .build();
+
+        TextInput transferQuestionInput = TextInput
+            .create(TRANSFER_QUESTION_INPUT_ID, "Transfer question menu", TextInputStyle.PARAGRAPH)
+            .setValue(originalMessage)
+            .setRequiredRange(3, 1000)
+            .build();
+
+        TextInput transferQuestionTag = TextInput
+            .create(TRANSFER_QUESTION_TAG, "Transfer question tags", TextInputStyle.SHORT)
+            .setValue("Java")
+            .build();
+
+
+
+        String transferQuestionModalComponentID = generateComponentId(author);
+        Modal reportModal = Modal.create(transferQuestionModalComponentID, "transfer question menu")
+            .addActionRow(transferQuestionTitle)
+            .addActionRow(transferQuestionInput)
+            .addActionRow(transferQuestionTag)
+            .build();
+
+        event.replyModal(reportModal).queue();
+    }
+
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        User author = event.getJDA().getUserById(args.get(0));
+        createForumPost(event, author);
+
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    public static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX
+                && !titleCompact.toLowerCase(Locale.US).contains("help");
+    }
+
+    public void createForumPost(ModalInteractionEvent event, User originalUser) {
+
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+
+        ForumChannel questionsForum = event.getJDA().getForumChannelById(QUESTIONS_CHANNEL_ID);
+
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+
+        ForumTag javaTag = tags.get(0);
+
+        event.getJDA()
+            .getForumChannelById(QUESTIONS_CHANNEL_ID)
+            .createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
```

Comment:
y always javaTag

Reply:
Earlier suggestion was to include a selection menu but didn't seam feasible, used java as it's default and most used tag. Plus this is just a feature to be used as "worst case scenario" clean up not to be used primarily. If any post do not belong to certain category, any helper can see that and change category.

Reply:
hmm, but this  won't work  anyways coz u can't just hard code the java tag, it is a config thing. The best option I can think of is having a selection menu

Reply:
u could read the config. the tags there are sorted by relevance, the first one is always considered the main mega tag

![config](https://i.imgur.com/OnmWAqs.png)

Reply:
this one is still not implemented, ill do this one hopefully soon.